schemas/2015-08-01/Microsoft.Compute.json

```diff
@@ -1,549 +1,835 @@
-                "properties": {
-                    "properties": {
-                        "publisher": {
-                            "type": "string"
-                        },
-                        "type": {
-                            "type": "string"
-                        },
-                        "typeHandlerVersion": {
-                            "type": "string"
-                        },
-                        "settings": {
-                            "type": "object"
-                        }
-                    },
-                    "type": "object",
-                    "required": [
-                        "publisher",
-                        "type",
-                        "typeHandlerVersion",
-                        "settings"
-                    ]
-                }
+        "apiVersion": {
+          "enum": [
+            "2015-05-01-preview",
+            "2015-06-15"
+          ]
+        },
+        "properties": {
+          "properties": {
+            "availabilitySet": {
+              "$ref": "#/definitions/id"
             },
-            "required": [
-                "type",
-                "apiVersion",
-                "properties"
+            "hardwareProfile": {
+              "$ref": "#/definitions/hardwareProfile"
+            },
+            "storageProfile": {
+              "$ref": "#/definitions/storageProfile"
+            },
+            "osProfile": {
+              "$ref": "#/definitions/osProfile"
+            },
+            "networkProfile": {
+              "$ref": "#/definitions/networkProfile"
+            }
+          },
+          "type": "object",
+          "required": [
+            "hardwareProfile",
+            "storageProfile",
+            "networkProfile"
+          ]
+        },
+        "resources": {
+          "type": "array",
+          "items": {
+            "allOf": [
+              {
+                "$ref": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#/definitions/resourceBase"
+              },
+              {
+                "oneOf": [
+                  {
+                    "$ref": "http://schema.management.azure.com/schemas/2014-10-01-preview/Microsoft.Authorization.json#/definitions/roleAssignments"
+                  },
+                  {
+                    "$ref": "http://schema.management.azure.com/schemas/2015-01-01/Microsoft.Authorization.json#/definitions/locks"
+                  },
+                  {
+                    "$ref": "http://schema.management.azure.com/schemas/2015-01-01/Microsoft.Resources.json#/definitions/links"
+                  },
+                  {
+                    "$ref": "#/definitions/extensionsChild"
+                  }
+                ]
+              }
             ]
+          },
+          "description": "Microsoft.Compute/virtualMachines: Resource Definition for Virtual Machines."
         }
+      },
+      "required": [
+        "properties"
```

Comment:
although the "apiVersion" and "type" is defined "required" in deploymentTemlate.json\resourceBase, we still suggest define them "required" here to help schema reader understand the requirement easier.